### PR TITLE
Using environment variables for configuration

### LINKED ISSUE #165: Twelve factor app guidelines: using environment variables for configuration
## What's missing?

The go app currently uses cobra to read a 'config' file for all its configuration settings. This config file contains configuration variables, and secrets, that a user cannot override easily using environment variable(s).

## Your preferred solution

https://12factor.net/config

Using APIGEE_X_Y_Z* environment variables to override settings would simplify deployment. Can the current implementation be extended with [viper](https://github.com/spf13/viper) to support this feature?

With Kubernetes it will also open the possibility to use separate Secret(s) and ConfigMap to store config variables.

### PR DESCRIPTION
**Pull request by rockspore**
Enabled env var binding for loading all configs except for the new envconfig.

Close #165

### CODE REVIEW

**Comment by theganyo** [comment]
Let's make sure our loading/resolving mechanism is documented at the top of the file.

File: config/config.go [hunk 1]
```diff
@@ -356,7 +402,29 @@ type StringTransformation struct {
 	Substitution string `yaml:"substitution,omitempty" json:"substitution,omitempty"`
 }
 
-// Load config
+// Load config with the given config file, secret paths and a flag specifying whether analytics credentials must be present.
```
Comment (rockspore) [inline config/config.go:405]: I added a chunk of documentation on `Load()` to describe how the config gets loaded. PTAL @theganyo

**Comment by theganyo** [comment]
General question: Is there a reason to use a combination of underscore and dot notation?

**Comment by rockspore** [comment]
> General question: Is there a reason to use a combination of underscore and dot notation?

Dot indicates separates structs, although this may not be super helpful to users.

**Comment by theganyo** [comment]
Makes sense... and we obviously can't use only underscores since we have some variable names with underscore. Would it make sense to go the other direction and only use dot notation?

**Comment by rockspore** [comment]
> Makes sense... and we obviously can't use only underscores since we have some variable names with underscore. Would it make sense to go the other direction and only use dot notation?

It this referring to the underscore after the `APIGEE` prefix?

**Comment by rockspore** [comment]
Sounds good.

**Review by theganyo — APPROVED** [review]

**Review by theganyo — APPROVED** [review]

**Pull request merged**